### PR TITLE
Expand README

### PR DESCRIPTION
**Pull request by ddebroy**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design

/kind documentation

> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Expand the usage section in the README

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,87 @@
-# csi-proxy
+# CSI Proxy
 
-CSI Proxy utility to enable CSI Plugins on Windows.
+CSI Proxy is a binary that exposes a set of gRPC APIs around storage operations
+over named pipes in Windows. A container, such as CSI node plugins, can mount
+the named pipes depending on operations it wants to exercise on the host and
+invoke the APIs.
+
+Each named pipe will support a specific version of an API (e.g. v1alpha1, v2beta1)
```
Comment (msau42) [inline README.md]: Does each api group have its own named pipe? ie disk v1alpha1, smb v2?

Comment (ddebroy) [reply README.md]: Correct. Example: https://github.com/kubernetes-csi/csi-proxy/blob/master/client/groups/filesystem/v1alpha1/client_generated.go

Comment (msau42) [reply README.md]: From the hostpath specified [below](https://github.com/kubernetes-csi/csi-proxy/pull/16/files#diff-04c6e90faac2675aa89e2176d2eec7d8R70), it's not clear to me how to distinguish between the named pipes for different api groups.

Is it not possible to serve multiple rpc services on a single named pipe?

**Pull request merged**